src/components/MultiSelect/MultiSelect.test.tsx

```diff
@@ -0,0 +1,720 @@
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import { axe } from "jest-axe";
+import * as React from "react";
+import renderer from "react-test-renderer";
+import MultiSelect from "./MultiSelect";
+import { MultiSelectItem } from "./MultiSelectTypes";
+
+const items = [
+  { id: "dogs", name: "Dogs" },
+  { id: "cats", name: "Cats" },
+  { id: "cars", name: "Cars" },
+  {
+    id: "colors",
+    name: "Colors",
+    children: [
+      { id: "red", name: "Red" },
+      { id: "blue", name: "Blue" },
+    ],
+  },
+  { id: "plants", name: "Plants" },
+  { id: "furniture", name: "Furniture" },
+];
+
+const MultiSelectDialogTestComponent = (componentId) => {
+  const [selectedItems, setSelectedItems] = React.useState({});
+
+  function handleChange(multiSelectId: string, itemId: string) {
```

Comment:
Copying previous comment since the first part was fixed but not the second:

> The handleChange and handleMixedStateChange functions are the same as the functions that are returned from the useMultiSelectHook. Can those be used instead of replicating the code here?